client/src/app/components/stock-stats/stock-stats.component.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Stock } from '../../interfaces/stock';
+import { MATERIAL_MODULE_DEPENDENCIES } from '../../shared.module';
+import { StockStatsComponent } from './stock-stats.component';
+
+// integration tests
+describe('StockStatsComponent', () => {
+  let component: StockStatsComponent;
+  let fixture: ComponentFixture<StockStatsComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: MATERIAL_MODULE_DEPENDENCIES,
+      declarations: [StockStatsComponent],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(StockStatsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
+
+const stockInfo: Stock = {
+  tickerSymbol: 'AC',
+  companyName: 'Air Canada',
+  industry: 'Transportation',
+  volatility: 'High',
+  stockChange: 4.27,
+  stockChangePercent: 1.68,
+  stockValue: 16.36,
+  logo:
+    'https://d1yjjnpx0p53s8.cloudfront.net/styles/logo-thumbnail/s3/0021/6847/brand.gif?itok=u0iVoArk',
+  stats: {
+    open: 100.5,
+    high: 200.2,
+    low: 516510,
+    volume: 151515,
+    mktCap: 51651,
+    wh: 455,
+    wl: 123,
+    avgVolume: 199410,
+  },
+};
+
+// unit tests
+describe('StockStats', () => {
+  let component: StockStatsComponent;
+
+  beforeEach(() => {
+    component = new StockStatsComponent();
+  });
+
+  it('should display correct stock open stat', () => {
+    component.stockInfo = stockInfo;
+    expect(component.open).toBe(component.stockInfo.stats.open.toFixed(2));
+    component.stockInfo.stats.open = stockInfo.stats.open;
+    expect(component.open).toBe(component.stockInfo.stats.open.toFixed(2));
+    component.stockInfo.stats.open = null;
+    expect(component.open).toBe('');
+    component.stockInfo = null;
+    expect(component.open).toBe('');
```

Comment:
I think if the value is not there it should return the string 'N/A' to show the user that the data loaded but isn't available. The test is written well, but we should change the logic